beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/publisher/BlockPublisherFulu.java

```diff
@@ -13,29 +13,46 @@
 
 package tech.pegasys.teku.validator.coordinator.publisher;
 
+import com.google.common.annotations.VisibleForTesting;
+import java.util.HashSet;
 import java.util.List;
+import java.util.OptionalInt;
+import java.util.Set;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import tech.pegasys.teku.ethereum.performance.trackers.BlockPublishingPerformance;
 import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
 import tech.pegasys.teku.networking.eth2.gossip.BlockGossipChannel;
 import tech.pegasys.teku.networking.eth2.gossip.DataColumnSidecarGossipChannel;
 import tech.pegasys.teku.spec.datastructures.blobs.DataColumnSidecar;
 import tech.pegasys.teku.spec.datastructures.blobs.versions.deneb.BlobSidecar;
 import tech.pegasys.teku.statetransition.blobs.RemoteOrigin;
 import tech.pegasys.teku.statetransition.block.BlockImportChannel;
+import tech.pegasys.teku.statetransition.datacolumns.CustodyGroupCountManager;
 import tech.pegasys.teku.validator.coordinator.BlockFactory;
 import tech.pegasys.teku.validator.coordinator.DutyMetrics;
 
 public class BlockPublisherFulu extends BlockPublisherPhase0 {
+  private static final Logger LOG = LogManager.getLogger();
 
   private final DataColumnSidecarGossipChannel dataColumnSidecarGossipChannel;
 
+  private final OptionalInt dasPublishWithholdColumnsEverySlots;
+
+  private UInt64 lastWithheldSlot = UInt64.MAX_VALUE;
```

Comment:
probably better to be an atomic reference to uint64... the reference can be final, and it would be more thread safe